### PR TITLE
chore: fix issues reported by `go vet`

### PR DESCRIPTION
**Pull request by mvertes**
Passing `go vet` is the minimum level of code quality for a go project. This is addressed here for the full mono-repo. Mainly trivial changes, except for a few `copy lock` issues which could be more meaningful (kept for a separate pull request).

Addresses #2954 (but not sufficient to close it yet).

Hint for reviewers: it's easier to review each commit individually.
 
<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Comment by thehowl** [comment]
Can you also add the `govet` linter to `.golangci.yml`?

**Comment by mvertes** [comment]
I will be happy to add `govet` to the CI, once we pass it 100%. Another PR is necessary to address the passing of Mutex by value, never a good sign, and more delicate to fix than the trivial changes here.

File: misc/logos/cmd/logos.go
Comment (thehowl) [inline misc/logos/cmd/logos.go:1]: I'll make a PR to nuke this directory

**Review by thehowl — APPROVED** [review]

**Comment by mvertes** [comment]
Ok. I see that there is already some comments `//nolint:govet` at the offender places (cough...), so yes, give it a try to govet in CI!

**Pull request merged**